langlib/lang.int/src/main/java/org/ballerinalang/langlib/integer/Sum.java

```diff
@@ -31,9 +39,14 @@
 
     public static long sum(long[] ns) {
         long sum = 0;
-        int size = ns.length;
-        for (int i = 0; i < size; i++) {
-            sum += ns[i];
+        for (long n : ns) {
+            try {
+                sum = Math.addExact(sum, n);
+            } catch (ArithmeticException e) {
+                throw ErrorCreator.createError(getModulePrefixedReason(INT_LANG_LIB,
+                        BallerinaErrorReasons.NUMBER_OVERFLOW_ERROR_IDENTIFIER),
+                        BLangExceptionHelper.getErrorDetails(RuntimeErrors.INT_RANGE_OVERFLOW_ERROR));
+            }
```

Comment:
We are doing this instead of calling `io.ballerina.runtime.internal.MathUtils#addExact` because we need to use a different error message (previously reason), right? Shall we add a comment mentioning this?

Alternatively, wondering if we should introduce overloaded methods in `MathUtils` and `ErrorUtils` to accept the error message.

@warunalakshitha, thoughts please?

My concern is that this should be the same as the additive expression. While it's highly unlikely that the addition logic would change, it would be better to have the logic in a single place IMO.

Reply:
+1

Reply:
Also, check whether there are any other lang lib methods, that have duplicate logic. I think core logic should be inside runtime classes not in lang lib.